charts/mongo-atlas/templates/scraper.yaml

```diff
@@ -0,0 +1,109 @@
+  retention:
+    {{- with .Values.scraper.retention.staleItemAge }}
+    staleItemAge: {{ . }}
+    {{- end }}
+  http:
+    - url: "https://cloud.mongodb.com/api/atlas/v1.0/groups/{{.Values.groupID}}/clusters"
+      digest: true
+      username:
+        valueFrom:
+          secretKeyRef:
+            name: "{{.Values.secret.name}}"
+            key: PUBLIC_KEY
+      password:
+        valueFrom:
+          secretKeyRef:
+            name: "{{.Values.secret.name}}"
+            key: PRIVATE_KEY
+      id: '$.id'
+      type: '$.configType'
+      name: '$.name'
+      createFields:
+        - $.createDate
+      properties:
+        - filter: config.configType == "Mongo::Replica"
+          name: 'URL'
+          links:
+            - label: Atlas
+              url: 'https://$(.config.name)'
+        - filter: config.configType == "Mongo::ReplicaSet"
+          name: 'Version'
+          text: '$(.config.mongoDBVersion)'
+        - filter: config.configType == "Mongo::ReplicaSet"
+          name: 'Region'
+          text: '$(.config.providerSettings.regionName)'
+      transform:
+        relationship:
+          - filter: config.configType == "Mongo::ReplicaSet"
+            expr: |
+              config.replicas.map(r, {
+                "external_id": r,
+              }).toJSON()
+        javascript: |+
+          var all = [];
+          for (var i = 0; i < config.results.length; i++) {
+            var cluster = config.results[i];
+
+            if (cluster.clusterType == "REPLICASET") {
+              cluster.configType = "Mongo::ReplicaSet";
+
+              var replicas = cluster.mongoURI.split(",");
+              var replicaHosts = [];
+              for (var j = 0; j < replicas.length; j++) {
+                var replica = replicas[j].trim().replace(/^.*:\/\//, '').split(/[/?#]/)[0].split(':')[0];
+                replicaHosts.push(replica);
+                all.push({
+                  "configType": "Mongo::Replica",
+                  "id": replica,
+                  "name": replica,
+                  "createDate": cluster.createDate,
+                })
+              }
+
+              cluster.replicas = replicaHosts;
+            }
+
+            all.push(cluster)
+          }
+
+          JSON.stringify(all)
+---
+{{- if .Values.scraper.slack.enabled}}
+apiVersion: configs.flanksource.com/v1
+kind: ScrapeConfig
+metadata:
+  name: "{{.Values.scraper.slack.name}}"
+spec:
+  schedule: {{ .Values.scraper.slack.schedule | quote }}
+  slack:
+    - channels: {{ .Values.scraper.slack.channels | toYaml | nindent 8 }}
+      since: "{{.Values.scraper.slack.since}}"
+      token:
+        valueFrom:
+          secretKeyRef:
+            name: {{.Values.scraper.slack.secretKeyRef.name}}
+            key: {{.Values.scraper.slack.secretKeyRef.key}}
+      rules:
+        - regexp: (?s):.*?\*(?P<status>Closed|Open)\*:.*?Query Targeting:(?P<message>.*?)<[^>]*\.net\|(?P<shard_url>[^>]+\.mongodb\.net)>
```

Comment:
Query targeting should be extracted as the change type - there could be many others